src/docfx/restore/RestoreMap.cs

```diff
@@ -193,17 +194,28 @@ string GetFileContentIfExists(string file)
         /// </summary>
         public static async Task<(string path, DependencyGit git)> TryGetGitRestorePath(string remote, string branch, string commit)
         {
-            var (path, slot) = await DependencySlotPool<DependencyGit>.TryGetSlot(remote, gits =>
-            {
-                var filteredGits = gits.Where(i => i.Branch == branch);
-                if (!string.IsNullOrEmpty(commit))
+            var restoreDir = AppData.GetGitDir(remote);
+
+            var (path, slot) = await DependencySlotPool<DependencyGit>.TryGetSlot(
+                remote,
+                async gits =>
                 {
-                    // found commit matched slot
-                    filteredGits = filteredGits.Where(i => i.Commit == commit);
-                }
+                    var filteredGits = gits.Where(i => i.Branch == branch);
 
-                return filteredGits.ToList();
-            });
+                    if (!string.IsNullOrEmpty(commit))
```

Comment:
suggestion:
 check commit first to return earlier?

Reply:
here we are assuming that we can get multiple slots with same commit.